clang/docs/ClangSYCLLinkWrapper.rst

```diff
@@ -0,0 +1,80 @@
+=======================
+Clang SYCL Link Wrapper
+=======================
+
+.. contents::
+   :local:
+
+.. _clang-sycl-link-wrapper:
+
+Introduction
+============
+
+This tool works as a wrapper around the SYCL device code linking process.
+The purpose of this wrapper is to provide an interface to link SYCL device
+bitcode in LLVM IR format, SYCL device bitcode in SPIR-V IR format, and native
+binary objects, and then use the SPIR-V LLVM Translator tool on fully linked
+device objects to produce the final output.
+After the linking stage, the fully linked device code in LLVM IR format may
+undergo several SYCL-specific finalization steps before the SPIR-V code
+generation step.
+The wrapper will also support the Ahead-Of-Time compilation flow. AOT
+compilation is the process of invoking the back-end at compile time to produce
+the final binary, as opposed to just-in-time (JIT) compilation when final code
+generation is deferred until application runtime time.
+
+Device code linking for SYCL offloading has several known quirks that
+makes it difficult to use in a unified offloading setting. Two of the primary
```

Comment:
```suggestion
make it difficult to use in a unified offloading setting. Two of the primary
```

Reply:
'makes' goes well with 'Device code linking'. WDYT?

Reply:
It reads interesting, as the association can also be with 'quirks'.  It is the quirks that make it difficult - it's not the device code linking that makes it difficult.